### PR TITLE
stats: Replace CPU 32/64 bit flag with GPU enumeration

### PR DESCRIPTION
**Pull request by MilhouseVH**
Replace CPU flag with GPU flag. Knowing if the CPU arch is 32-bit or 64-bit is now largely redundant information. Having a handle on how many Nvidia or AMD or Intel GPU installations we have is far more relevant/useful.

Much of this is blind, due to lack of hardware. I've tested it with i915-based Skylake NUC and ION2 based Revo3700 (Nvidia-legacy). VM (vmwgfx) should also be working. I haven't tested with any AMD hardware.

This may not work reliably with multi-GPU setups - hopefully it will just default to the first connected GPU.

To test:
1. enable `debugging` in kernel config
2. `grep -E "check_updates_v2|load_value|get_gpu" /storage/.kodi/temp/kodi.log`
```
18:11:14.627 T:140215976978176   DEBUG: ## LibreELEC Addon ## system::load_values ## enter_function
18:11:14.627 T:140215976978176   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## Using card: card0
18:11:14.669 T:140215976978176   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## Detected gpu driver: i915
18:11:14.669 T:140215976978176   DEBUG: ## LibreELEC Addon ## system::load_values ## loaded gpu_type 1
18:11:18.592 T:140215976978176   DEBUG: ## LibreELEC Addon ## system::load_values ## exit_function
18:11:18.638 T:140215202989824   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## enter_function
18:11:18.638 T:140215202989824   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## URL: https://update.libreelec.tv/updates.php?i=4fd0f16f4d2503cb9f6651f37d273328&d=LibreELEC&pa=Generic.x86_64&v=devel-20180531173100-#0531b-ge1f38c5&l=1
18:11:18.639 T:140215976978176   DEBUG: ## LibreELEC Addon ## connman::load_values ## enter_function
18:11:18.645 T:140215976978176   DEBUG: ## LibreELEC Addon ## connman::load_values ## exit_function
18:11:18.645 T:140215976978176   DEBUG: ## LibreELEC Addon ## services::load_values ## enter_function
18:11:18.645 T:140215976978176   DEBUG: ## LibreELEC Addon ## services::load_values ## exit_function
18:11:18.702 T:140215202989824   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## RESULT: ''
18:11:18.702 T:140215202989824   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## exit_function
```

### CODE REVIEW

File: src/resources/lib/modules/system.py [hunk 1]
```diff
@@ -341,13 +340,59 @@ def exit(self):
...
+        gpu_drivers = {'i915': 1, 'i965': 2,
+                       'nvidia': 3, 'nvidia-legacy': 4,
+                       'r300': 5, 'r600': 6, 'radeonsi': 7,
+                       'vmwgfx': 8, 'virtio_gpu': 9}
+
+        gpu_card = self.get_gpu_card()
+        self.oe.dbg_log('system::get_gpu_type', 'Using card: %s' % gpu_card, 0)
+
+        if gpu_card:
+            gpu_driver = self.oe.execute('/usr/bin/udevadm info --name=/dev/dri/%s --attribute-walk 2>/dev/null | /usr/bin/grep -m1 DRIVERS= | /usr/bin/sed \'s/["=]/ /g\' | /usr/bin/awk \'{print $2}\'' % gpu_card, get_result=1)
```
Comment (lrusak) [inline src/resources/lib/modules/system.py]: this is pretty ugly, can we not just get the info from udevadm and parse in python?

Comment (MilhouseVH) [reply src/resources/lib/modules/system.py]: Yes it's ugly, but the output of `udevadm` isn't particularly amenable to Python processing - it's not JSON, unfortunately:

```
LibreELEC:~ # udevadm info --name=/dev/dri/card0 --attribute-walk

Udevadm info starts with the device specified by the devpath and then
walks up the chain of parent devices. It prints for every device
found, all possible attributes in the udev rules key format.
A rule to match, can be composed by the attributes of the device
and the attributes from one single parent device.

  looking at device '/devices/pci0000:00/0000:00:1c.0/0000:01:00.0/drm/card0':
    KERNEL=="card0"
    SUBSYSTEM=="drm"
    DRIVER==""

  looking at parent device '/devices/pci0000:00/0000:00:1c.0/0000:01:00.0':
    KERNELS=="0000:01:00.0"
    SUBSYSTEMS=="pci"
    DRIVERS=="nvidia"
    ATTRS{boot_vga}=="1"
    ATTRS{broken_parity_status}=="0"
    ATTRS{class}=="0x030000"
    ATTRS{consistent_dma_mask_bits}=="32"
    ATTRS{current_link_speed}=="2.5 GT/s"
    ATTRS{current_link_width}=="1"
    ATTRS{d3cold_allowed}=="1"
    ATTRS{device}=="0x0a64"
    ATTRS{dma_mask_bits}=="40"
    ATTRS{driver_override}=="(null)"
    ATTRS{enable}=="2"
    ATTRS{irq}=="30"
    ATTRS{local_cpulist}=="0-3"
    ATTRS{local_cpus}=="f"
    ATTRS{max_link_speed}=="5 GT/s"
    ATTRS{max_link_width}=="16"
    ATTRS{msi_bus}=="1"
    ATTRS{numa_node}=="-1"
    ATTRS{revision}=="0xa2"
    ATTRS{subsystem_device}=="0x047a"
    ATTRS{subsystem_vendor}=="0x1025"
    ATTRS{vendor}=="0x10de"

  looking at parent device '/devices/pci0000:00/0000:00:1c.0':
    KERNELS=="0000:00:1c.0"
    SUBSYSTEMS=="pci"
    DRIVERS=="pcieport"
    ATTRS{broken_parity_status}=="0"
    ATTRS{class}=="0x060400"
    ATTRS{consistent_dma_mask_bits}=="32"
    ATTRS{current_link_speed}=="2.5 GT/s"
    ATTRS{current_link_width}=="1"
    ATTRS{d3cold_allowed}=="1"
    ATTRS{device}=="0x27d0"
    ATTRS{dma_mask_bits}=="32"
    ATTRS{driver_override}=="(null)"
    ATTRS{enable}=="1"
    ATTRS{irq}=="24"
    ATTRS{local_cpulist}=="0-3"
    ATTRS{local_cpus}=="f"
    ATTRS{max_link_speed}=="2.5 GT/s"
    ATTRS{max_link_width}=="1"
    ATTRS{msi_bus}=="1"
    ATTRS{numa_node}=="-1"
    ATTRS{revision}=="0x02"
    ATTRS{secondary_bus_number}=="1"
    ATTRS{subordinate_bus_number}=="1"
    ATTRS{subsystem_device}=="0x047a"
    ATTRS{subsystem_vendor}=="0x1025"
    ATTRS{vendor}=="0x8086"

  looking at parent device '/devices/pci0000:00':
    KERNELS=="pci0000:00"
    SUBSYSTEMS==""
    DRIVERS==""

```

**Comment by MilhouseVH** [comment]
@lrusak - updated, more Pythonic?

With `debugging` enabled:
Nvidia (legacy):
```
20:17:50.057 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::load_values ## enter_function
20:17:50.058 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## Using card: card0
20:17:50.484 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu path: /devices/pci0000:00/0000:00:1c.0/0000:01:00.0/drm/card0
20:17:50.484 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## driver path: /devices/pci0000:00/0000:00:1c.0/0000:01:00.0
20:17:50.484 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu props: {'SUBSYSTEM': 'pci', 'ID_VENDOR_FROM_DATABASE': 'NVIDIA Corporation', 'PCI_CLASS': '30000', 'ID_PCI_INTERFACE_FROM_DATABASE': 'VGA controller', 'ID_PCI_SUBCLASS_FROM_DATABASE': 'VGA compatible controller', 'MODALIAS': 'pci:v000010DEd00000A64sv00001025sd0000047Abc03sc00i00', 'ID_MODEL_FROM_DATABASE': 'GT218 [ION]', 'DEVPATH': '/devices/pci0000:00/0000:00:1c.0/0000:01:00.0', 'DRIVER': 'nvidia', 'PCI_ID': '10DE:0A64', 'ID_PCI_CLASS_FROM_DATABASE': 'Display controller', 'PCI_SUBSYS_ID': '1025:047A', 'USEC_INITIALIZED': '12986208', 'PCI_SLOT_NAME': '0000:01:00.0'}
20:17:50.484 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu driver: nvidia
20:17:50.489 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::load_values ## loaded gpu_type 4
20:17:56.269 T:139629655226112   DEBUG: ## LibreELEC Addon ## system::load_values ## exit_function
20:17:56.466 T:139629655226112   DEBUG: ## LibreELEC Addon ## connman::load_values ## enter_function
20:17:56.466 T:139629739087616   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## enter_function
20:17:56.467 T:139629739087616   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## URL: https://update.libreelec.tv/updates.php?i=c9292fe864dd44e887a3b3593db8c7b4&d=LibreELEC&pa=Generic.x86_64&v=devel-20180531201310-#0531e-ge1f38c5&l=4
20:17:56.510 T:139629655226112   DEBUG: ## LibreELEC Addon ## connman::load_values ## exit_function
20:17:56.511 T:139629655226112   DEBUG: ## LibreELEC Addon ## services::load_values ## enter_function
20:17:56.549 T:139629655226112   DEBUG: ## LibreELEC Addon ## services::load_values ## exit_function
20:17:56.589 T:139629739087616   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## RESULT: ''
20:17:56.589 T:139629739087616   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## exit_function
```

NUC (Skylake):
```
20:15:39.258 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::load_values ## enter_function
20:15:39.259 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## Using card: card0
20:15:39.314 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu path: /devices/pci0000:00/0000:00:02.0/drm/card0
20:15:39.315 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## driver path: /devices/pci0000:00/0000:00:02.0
20:15:39.315 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu props: {'SUBSYSTEM': 'pci', 'ID_VENDOR_FROM_DATABASE': 'Intel Corporation', 'PCI_CLASS': '30000', 'ID_PCI_INTERFACE_FROM_DATABASE': 'VGA controller', 'ID_PCI_SUBCLASS_FROM_DATABASE': 'VGA compatible controller', 'MODALIAS': 'pci:v00008086d00001926sv00008086sd00002063bc03sc00i00', 'ID_MODEL_FROM_DATABASE': 'Iris Graphics 540', 'DEVPATH': '/devices/pci0000:00/0000:00:02.0', 'DRIVER': 'i915', 'PCI_ID': '8086:1926', 'xorg_driver': 'i915', 'ID_PCI_CLASS_FROM_DATABASE': 'Display controller', 'PCI_SUBSYS_ID': '8086:2063', 'USEC_INITIALIZED': '2824573', 'SYSTEMD_WANTS': 'xorg-configure@i915.service', 'TAGS': ':systemd:', 'PCI_SLOT_NAME': '0000:00:02.0'}
20:15:39.315 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::get_gpu_type ## gpu driver: i915
20:15:39.315 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::load_values ## loaded gpu_type 1
20:15:43.947 T:140322134808320   DEBUG: ## LibreELEC Addon ## system::load_values ## exit_function
20:15:43.999 T:140321489094400   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## enter_function
20:15:43.999 T:140321489094400   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## URL: https://update.libreelec.tv/updates.php?i=4fd0f16f4d2503cb9f6651f37d273328&d=LibreELEC&pa=Generic.x86_64&v=devel-20180531201310-#0531e-ge1f38c5&l=1
20:15:44.000 T:140322134808320   DEBUG: ## LibreELEC Addon ## connman::load_values ## enter_function
20:15:44.008 T:140322134808320   DEBUG: ## LibreELEC Addon ## connman::load_values ## exit_function
20:15:44.008 T:140322134808320   DEBUG: ## LibreELEC Addon ## services::load_values ## enter_function
20:15:44.009 T:140322134808320   DEBUG: ## LibreELEC Addon ## services::load_values ## exit_function
20:15:44.081 T:140321489094400   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## RESULT: ''
20:15:44.082 T:140321489094400   DEBUG: ## LibreELEC Addon ## system::check_updates_v2 ## exit_function
```

**Comment by MilhouseVH** [comment]
Added a "stats opt-out" setting, passing "DONOTRACK" to update server which already supports this facility, no need to hack machine-id. Enabled by default.

s1

**Comment by MilhouseVH** [comment]
Cosmetic layout tweaks - I noticed we are clipping help text in a few places. Aligning the horizontal separator in the main list with the same separator in the menu (above the logo) looks better (IMHO) and gives us a bit more room for help text. The wizard also had some issues with positioning, so fixed those too.

Wizard before/after - increase space for text/reduce button space:
s2b
s2a

Give title "The following Networks are currently available" a bit more space between the paragraph above
s3b
s3a

s4b
s4a

Main settings before/after - align horizontal separators:

"About" is still OK:
s5b
s6b

Avoid "help" text clipping (descenders are being clipped):
s6b
s6a

File: language/English/strings.po [hunk 2]
```diff
@@ -312,6 +316,10 @@ msgctxt "#32020"
 msgid "Available Versions"
 msgstr ""
 
+msgctxt "#32021"
+msgid "Submit statistics"
```
Comment (lrusak) [inline language/English/strings.po]: Statistics

Comment (MilhouseVH) [reply language/English/strings.po]: Thanks, updated.

**Pull request merged**